### PR TITLE
[SYCL][ESIMD] Add support for named barrier APIs

### PR DESCRIPTION
**Pull request by sndmitriev**
Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp [hunk 1]
```diff
@@ -885,6 +889,34 @@ static void translateUnPackMask(CallInst &CI) {
   CI.replaceAllUsesWith(TransCI);
 }
 
+// This function sets VCNamedBarrierCount attribute to set
+// the number of named barriers required by a kernel
+static void translateNbarrierInit(CallInst &CI) {
+  auto F = CI.getParent()->getParent();
```
Comment (mlychkov) [inline llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp:895]:
```suggestion
  auto *F = CI.getFunction();
```

**Review by mlychkov — DISMISSED** [review]
Other than a small nit LGTM.

**Review by kbobrovs — APPROVED** [review]
Please also don't forget to add an E2E test.

**Comment by sndmitriev** [comment]
Errors on buildbot/Lit_With_Cuda look like an infrastructure problem.

**Comment by kbobrovs** [comment]
@intel/dpcpp-tools-reviewers, ping

**Review by mlychkov — APPROVED** [review]

**Pull request merged**